packages/common/src/templates/HubTemplate.ts

```diff
@@ -0,0 +1,225 @@
+import { IArcGISContext } from "../ArcGISContext";
+import { getEditorConfig } from "../core/schemas/getEditorConfig";
+import { IEditorConfig } from "../core/behaviors/IWithEditorBehavior";
+import { HubItemEntity } from "../core/HubItemEntity";
+import { IHubTemplate, IHubTemplateEditor } from "../core/types/IHubTemplate";
+import { TemplateEditorType } from "./_internal/TemplateSchema";
+import { DEFAULT_TEMPLATE } from "./defaults";
+import {
+  createTemplate,
+  deleteTemplate,
+  editorToTemplate,
+  updateTemplate,
+} from "./edit";
+import { fetchTemplate } from "./fetch";
+import { IEntityEditorContext } from "../core/types";
+import { enrichEntity } from "../core/enrichEntity";
+import { cloneObject } from "../util";
+
+/** Hub Template Class */
```

Comment:
Maybe add more explanation here that this is used for all Solution items despite the Hub team not "owning" that item type. Main goal is to facilitate editing the meta info (the "item"), managing sharing etc